django/db/models/expressions.py

```diff
@@ -1021,11 +1021,13 @@ class Subquery(Expression):
     def __init__(self, queryset, output_field=None, **extra):
         self.query = queryset.query
         self.extra = extra
+        # Prevent the QuerySet from being evaluated.
+        self.queryset = queryset._chain(_result_cache=[], prefetch_done=True)
         super().__init__(output_field)
 
     def __getstate__(self):
         state = super().__getstate__()
-        state.pop('_constructor_args', None)
+        state['_constructor_args'] = (self.queryset,), {}
```

Comment:
One thing that worries me here is that it'll discard any `kwargs` so if `output_field` is provided, since we allow it to be passed as a positional argument it will be lost. e.g. `Subquery(queryset, models.BooleanField())`.

Reply:
True, I pushed edits to preserve them.